node/pkg/adminrpc/adminserver.go

```diff
@@ -1162,7 +1162,18 @@ func (s *nodePrivilegedService) SignExistingVAA(ctx context.Context, req *nodev1
 	}
 
 	// Add local signature
-	newVAA.AddSignature(s.gk, uint8(localGuardianIndex))
+	sig, err := s.guardianSigner.Sign(v.SigningDigest().Bytes())
+	if err != nil {
+		panic(err)
+	}
+
+	sigData := [65]byte{}
```

Comment:
I like how these lines are captured by `AddSignature` in the existing implementation. Any thoughts on putting this code into a method on the VAA rather than modifying it directly?

Reply:
Going to resolve this comment and make a note in the PR description about this decision in particular. I'm keen to get WH feedback on this.